insights/parsers/krb5.py

```diff
@@ -47,62 +50,104 @@
-
+class Krb5ConfigParser(Parser):
     def parse_content(self, content):
         dict_all = {}
         is_squ = False
         section_name = ""
         squ_value = {}
         squ_section_name = ""
         section_value = {}
-        for line in content:
-            """Get active line """
-            if "includedir" in line or line.startswith("#") or not line:
+        for line in get_active_lines(content):
+            if "includedir" in line:
                 continue
-            """[XXX] means a section, get the section name and the value format is dict."""
-            if "[" in line and "]" in line:
-                """If first section, just get the section name, if not, set the value to the former section and get the section name"""
-                if not section_name:
-                    section_name = line.strip().strip("[]")
-                    section_value = {}
-                else:
+            # [XXX] means a section, get the section name and the value format is dict.
+            if line.startswith("[") and line.endswith("]"):
+                # If first section, just get the section name, if not, set the value to the former section and get the section name
+                if section_name:
                     dict_all[section_name] = section_value
-                    section_name = line.strip().strip("[]")
-                    section_value = {}
+                section_name = line.strip().strip("[]")
+                section_value = {}
                 continue
-            """Common key value format is XXX = YYY"""
+            # Common key value format is XXX = YYY
             if "=" in line and "{" not in line and "}" not in line and not is_squ:
                 key_value = line.strip().split("=")
                 section_value[key_value[0].strip()] = key_value[1].strip()
                 continue
-            """"Special key value format is XXX = YYY{%ZZZ}"""
+            # Special key value format is XXX = YYY{%ZZZ}
             if "=" in line and "{" in line and "}" in line and len(line.split("=")) == 2 and not is_squ:
                 key_value = line.strip().split("=")
                 section_value[key_value[0].strip()] = key_value[1].strip()
                 continue
-            """If in {} sub_section, get the key_value pair"""
+            # If in {} sub_section, get the key_value pair
             if "=" in line and is_squ:
                 key_value = line.strip().split("=")
-                squ_value[key_value[0].strip()] = key_value[1].strip()
+                key = key_value[0].strip()
+                value = key_value[1].strip()
+                if key in squ_value:
+                    if isinstance(squ_value[key], list):
+                        squ_value[key].append(value)
+                    else:
+                        squ_value[key] = [squ_value[key]]
+                        squ_value[key].append(value)
+                else:
+                    squ_value[key] = value
```

Comment:
~~~
if key in squ_value:
       if isinstance(squ_value[key], str):
            squ_value[key] = [squ_value[key]]
       squ_value[key].append(value)
~~~